File: Assets/Bundles/Server/Resources/TS/Services/Global/Chat/Commands/DieCommand.lua
-- Compiled with unity-ts v2.1.0-75
local Flamework = require("Shared/rbxts_include/node_modules/@easy-games/flamework-core/out/init").Flamework
local ChatCommand = require("Server/TS/Services/Global/Chat/Commands/ChatCommand").ChatCommand
local DieCommand
do
	local super = ChatCommand
	DieCommand = setmetatable({}, {
		__tostring = function()
			return "DieCommand"
		end,
		__index = super,
	})
	DieCommand.__index = DieCommand
	function DieCommand.new(...)
		local self = setmetatable({}, DieCommand)
		return self:constructor(...) or self
	end
	function DieCommand:constructor()
		super.constructor(self, "die")
	end
	function DieCommand:Execute(player, args)
		-- Fetch target entity.
		local target = (Flamework.resolveDependency("Bundles/Server/Services/Global/Entity/EntityService@EntityService")):GetEntityByClientId(player.clientId)
		-- Handle invalid entity.
		if target == nil then
			player:SendMessage("invalid target")
			return nil
		end
		-- Kill entity.
		(Flamework.resolveDependency("Bundles/Server/Services/Global/Damage/DamageService@DamageService")):InflictDamage(target, math.huge, {
			ignoreCancelled = true,
		})
	end
end
return {
	DieCommand = DieCommand,
}
-- ----------------------------------
-- ----------------------------------

File: Assets/Typescript~/src/Server/Services/Global/Chat/Commands/DieCommand.ts
import { Dependency } from "@easy-games/flamework-core";
import { ChatCommand } from "Imports/Core/Shared/Commands/ChatCommand";
import { DamageService } from "Server/Services/Global/Damage/DamageService";
import { EntityService } from "Server/Services/Global/Entity/EntityService";
import { Player } from "Shared/Player/Player";

export class DieCommand extends ChatCommand {
	constructor() {
		super("die");
	}

	public Execute(player: Player, args: string[]): void {
		/* Fetch target entity. */
		const target = Dependency<EntityService>().GetEntityByClientId(player.clientId);

		/* Handle invalid entity. */
		if (target === undefined) {
			player.SendMessage("invalid target");
			return;
		}

		/* Kill entity. */
		Dependency<DamageService>().InflictDamage(target, math.huge, {
			ignoreCancelled: true,
		});
	}
}
